bin/check-s3-object.rb

```diff
@@ -159,17 +159,10 @@ def run
         key_search = config[:key_prefix]
         output = s3.list_objects(bucket: config[:bucket_name], prefix: key_search)
 
-        if output.contents.size.to_i < 1
-          critical "Object with prefix \"#{key_search}\" not found in bucket #{config[:bucket_name]}"
-        end
-
-        if output.contents.size.to_i > 1
```

Comment:
Hmm I am not sure how I feel about this, on the one hand it makes total sense what you want to do but I also see a use case where someone creates something that is unexpected and you want to be alerted on it. I think the best thing would be to offer an option such as `--crit-on-multiple-objects` with a default of true and then you just add that condition to this one. This makes it backwards compatible and adds the functionality you want.

Reply:
Actually I thought about this more and even then the question od what is the latest object is a rather ambiguous question. Say you are storing logs on s3 with a standard `logrotate` style so the latest artifact would be `syslog.log` and the next day the newest file would still be `syslog.log` renaming the old file to `syslog.log.1` this might have been the original intent in why they said you must only match a single key. If we want to add this we need to add the option mentioned above and specify if you would like to check the oldest or newest object based on `last_modified`.

Reply:
Thanks for the feedback @majormoses, I will give this some more thought and take another shot at it.